### PR TITLE
[hotstar] added support for downloading playlist

### PR DESCRIPTION
**Pull request by alpeshvalia**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [ ] New extractor
- [x] New feature

---

### Description of your *pull request* and other information

Support for download all episodes(playlist) from HotStar.
Steps to get playlist url:
- Open any show
- Click on 'Episodes'
- Copy the url
- E.g http://www.hotstar.com/tv/pow-bandi-yuddh-ke/10999/episodes/10856/9993

Hotstar works with --prefer-ffmpeg option only. This is an existing issue with single video download also.

### CODE REVIEW

File: youtube_dl/extractor/hotstar.py [hunk 1]
```diff
@@ -99,3 +100,59 @@ def _real_extract(self, url):
             'episode_number': int_or_none(video_data.get('episodeNumber')),
             'series': video_data.get('contentTitle'),
         }
+
+class HotStarBaseIE(InfoExtractor):
```
Comment (dstftw) [inline youtube_dl/extractor/hotstar.py]: What's the point of this base class? It's only inherited once.

File: youtube_dl/extractor/hotstar.py [hunk 2]
```diff
@@ -99,3 +100,59 @@ def _real_extract(self, url):
...
+    @classmethod
+    def _extract_url_info(cls, url):
```
Comment (dstftw) [inline youtube_dl/extractor/hotstar.py]: What's the point of this method? It's only used once.

File: youtube_dl/extractor/hotstar.py [hunk 3]
```diff
@@ -99,3 +100,59 @@ def _real_extract(self, url):
...
+    }, {
+        'url': 'http://www.hotstar.com/tv/pow-bandi-yuddh-ke/10999/episodes/10856/9993',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        series_id, playlist_id, playlist_title = self._extract_url_info(url)
+
+        collection = self._download_json(
+            "http://search.hotstar.com/AVS/besc?action=SearchContents&appVersion=5.0.39&channel=PCTV&moreFilters=series:%s;&query=*&searchOrder=last_broadcast_date+desc,year+asc,title+asc&type=EPISODE" % playlist_id,
```
Comment (dstftw) [inline youtube_dl/extractor/hotstar.py]: Query should be passed as `query` parameter.

File: youtube_dl/extractor/hotstar.py [hunk 4]
```diff
@@ -99,3 +100,59 @@ def _real_extract(self, url):
...
+class HotStarBaseIE(InfoExtractor):
+    @classmethod
+    def _extract_url_info(cls, url):
+        mobj = re.match(cls._VALID_URL, url)
+        return mobj.group('series_id'), mobj.group('playlist_id'), mobj.group('playlist_title')
+
+    def _extract_from_json_url(self, series_id, playlist_title, video ):
+
+        picture_url = video.get('urlPictures');
+        thumbnail = 'http://media0-starag.startv.in/r1/thumbs/PCTV/%s/%s/PCTV-%s-hs.jpg' % ( picture_url[-2:], picture_url, picture_url )
```
Comment (dstftw) [inline youtube_dl/extractor/hotstar.py]: Will break if `picture_url` is `None`.

File: youtube_dl/extractor/hotstar.py [hunk 5]
```diff
@@ -99,3 +100,59 @@ def _real_extract(self, url):
...
+
+        episode_title = video.get('episodeTitle')
+        episode_title = episode_title.lower().replace(' ', '-')
```
Comment (dstftw) [inline youtube_dl/extractor/hotstar.py]: Will break if `episode_title` is `None`.

**Comment by dstftw** [comment]
1. Fix issues.
2. Check code with flake8.
3. Squash commits.
4. Provide reliable evidence that the original code is in public domain or released under Unlicense.

File: youtube_dl/extractor/hotstar.py (hunk 3 above)
Comment (alpeshvalia) [inline youtube_dl/extractor/hotstar.py]: Tried to use it but this seems to be encoded. server is not able to accept encoded params. let me know if there is way not to encode this url

**Comment by alpeshvalia** [comment]
Plz let me know how to squash... never did it

**Comment by alpeshvalia** [comment]
Plz dnt merge it, i wl create new pull request as some commits were made from different email

**Pull request closed without merging**